From f945861fb8f3b55563ca0225b5ed41ca737b96a9 Mon Sep 17 00:00:00 2001
From: Guilin Zhong <guilin.zhong@intel.com>
Date: Thu, 14 Aug 2014 19:02:19 +0800
Subject: [PATCH 170/176] atomisp2: refcount is not clean in isp timeout
 recovery

Sometimes ov680 is not streaming on, but it creates streams
and creates pipes in atomisp_set_fmt function.

Meanwhile, ov8858 triggered isp timeout. Streams
and pipes need to be destroyed in isp timeout recovery.

Depends on: I2df711d83f2ae7b556b546df85932c0a7c8b289d
Depends on: I797816357c0dadc5c66b972d7b253ac1b96a5309

Change-Id: I3d7c4f28567065b655399eb353766057d8102f14
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-7681
Signed-off-by: Guilin Zhong <guilin.zhong@intel.com>
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/28018
Reviewed-by: Soren Friis <soren.friis@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Reviewed-by: Jukka O Kaartinen <jukka.o.kaartinen@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c    | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 64de1c6..4aec89e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1244,8 +1244,8 @@ static void __atomisp_css_recover(struct atomisp_device *isp)
 	for (i = 0; i < isp->num_of_streams; i++) {
 		struct atomisp_sub_device *asd = &isp->asd[i];
 
-		if (asd->streaming !=
-				ATOMISP_DEVICE_STREAMING_ENABLED)
+		if (asd->streaming != ATOMISP_DEVICE_STREAMING_ENABLED &&
+		    !asd->stream_prepared)
 			continue;
 
 		depth_cnt++;
-- 
1.9.1

